mac/VibeTunnel/Core/Services/SessionMonitor.swift

```diff
@@ -145,6 +175,35 @@ final class SessionMonitor {
 
             self.sessions = sessionsDict
             self.lastError = nil
+
+            // Notify for sessions that have just ended
+            if firstFetchDone && UserDefaults.standard.bool(forKey: "showNotifications") {
+                let ended = Self.detectEndedSessions(from: oldSessions, to: sessionsDict)
```

Comment:
i wonder if this wouldn't be better in server and then send notifications via socket.